include/boost/graph/r_c_shortest_paths.hpp

```diff
@@ -21,12 +23,12 @@ namespace boost {
 
 // r_c_shortest_paths_label struct
 template<class Graph, class Resource_Container>
-struct r_c_shortest_paths_label
+struct r_c_shortest_paths_label : public boost::enable_shared_from_this<r_c_shortest_paths_label<Graph, Resource_Container> >
```

Comment:
I've never used `enable_shared_from_this`; what does it do?

Reply:
[enable_shared_from_this](https://stackoverflow.com/questions/712279/what-is-the-usefulness-of-enable-shared-from-this) is a C++11 construct. See the link, it is better explained than I could.